### PR TITLE
Grafana UI: Menu - Add autofocus by default to menu component

### PR DESCRIPTION
**Pull request by axelavargas**
## Problem:
Dropdown menus containing async items (specifically, `PanelMenu` from `PanelChrome`) do not trigger autofocus to the container. When we don't have autofocus the logic from the `Menu` component to handle keyboard navigation does not work.

_This issue is coming from rendering panels using scenes, we noticed the keyboard does not work, the tricky part was to understand why it was working in the old architecture  (core grafana) and not using scenes._

<table>
<tr>
 <td> Bug
 <td> Fix
<tr>
 <td> 

image
 
<td> 

MenuScenesWorking

</table>

####  Old Architecture Behavior:

- Panel Rendering: `PanelChrome` renders a menu using `PanelMenu`. This dropdown, when visible, renders a menu provided by `PanelStateWrapper -->PanelHeaderMenu`.
- Menu Interaction: When a user clicks the menu, the dropdown's visibility changes.
- Menu Rendering: then `PanelHeaderMenu` is triggered, initially rendering an empty Menu (`<div tabindex="-1" role="menu"...`).
- Autofocus Triggering: The Autofocus (from React Aria) in the Dropdown is activated, focusing on the role="menu" element.
- Item Availability: Once items become available, PanelHeaderMenu re-renders with items. However, the focus remains on the container with role="menu".

#### New Architecture Behavior:

- Panel Rendering: Similar to the old architecture, `PanelChrome` uses `PanelMenu`. However, this time it's coupled with `VizPanelMenu -> VizPanelMenuRenderer.`
- Menu Interaction: When a user clicks the menu, `VizPanelMenu` get activated,  which in turn calls `panelMenuBehavior.`
- Asynchronous Item Gathering: Items are loaded asynchronously.
- Autofocus Triggering: The Autofocus (from React Aria) in the `Dropdown` activates, but at this point, there are no focusable elements.
- Delayed Menu Rendering: Once items are ready, `VizPanelMenuRenderer` is triggered, rendering the Menu with items. However, the container lacks focus.

In short: The problem is the order of execution, in the case of Scenes, the autofocus from react aria executes right after the `panelMenuBehavior `, but before the `VizPanelMenuRenderer`, this means the container of the menu never gets the focus.

## Proposal:

This PR adds by default autofocus to the `Menu` component, but also offers a possibility to override that prop for cases where autofocus is not desired. 
I'm open to suggestions and welcome any feedback or alternative ideas.

### CODE REVIEW

**Comment by torkelo** [comment]
Really great work figuring this out! 

I tried fixing it by always rendering an empty Menu but for me this does not fix the problem 
https://github.com/grafana/scenes/pull/483

**Review by dprokop — APPROVED** [review]
Nice investigation @axelavargas! I'm in favor in your fix instead of the one proposed by @torkelo as yours is fixing it for other that may be using Dropdown/Menu in a similar way we use it in Scenes.

**Review by torkelo — COMMENTED** [review]
@dprokop @axelavargas I am also in favor of this fix as this fix actually fixes the issue (mine alternative did not). 

The thing that I am not 100% about is that this fix requires a new prop and always adds a new FocusScope here that is 99% time not needed. 

I think the Scenes scenario of rendering a react component in the Dropdown overlay that initially on first render renders nothing (null), in the SceneComponentWrapper is very unique, I don't think this scenario is going to be common (we have yet to see it in any other scenario in core).

So the ideal fix would be to wrap this part in a FocusScope, https://github.com/grafana/scenes/blob/main/packages/scenes/src/components/VizPanel/VizPanelMenu.tsx#L53 

but then we have to add a dependency to react-aria. 

Updated https://github.com/grafana/scenes/pull/483 with a manual focus call (same logic that FocusScope autoFocus use). With that solution we do not need to complicate the Menu component with a new prop and FocusScope.

**Comment by axelavargas** [comment]
@torkelo Yes, good point here, the use case of having a dropdown menu with async items is quite unique. I think we can go with your proposal, I am also hesitant about adding this autofocus by default.

**Comment by axelavargas** [comment]
Closed in favor https://github.com/grafana/scenes/pull/483

**Comment by torkelo** [comment]
@axelavargas it's not the async items that is unique to the scenes scenario (the core scenario also has async items). It's the fact that in scenes the Overlay body renders no Menu on the first render, VizPanelMenu renders null on the first render, and we can't change that as all scene react components do that before they are activated. 

So it's not the async rendering of items, but the whole menu :)
